tests/testTriggers.go

```diff
@@ -10962,6 +10976,237 @@ func TriggerResetDiscardMounts(contexts *[]*scheduler.Context, recordChan *chan
 	})
 }
 
+func TriggerScaleFADAVolumeAttach(contexts *[]*scheduler.Context, recordChan *chan *EventRecord) {
+	defer endLongevityTest()
+	startLongevityTest(ScaleFADAVolumeAttach)
+	defer ginkgo.GinkgoRecover()
+	var wg sync.WaitGroup
+
+	event := &EventRecord{
+		Event: Event{
+			ID:   GenerateUUID(),
+			Type: ScaleFADAVolumeAttach,
+		},
+		Start:   time.Now().Format(time.RFC1123),
+		Outcome: []error{},
+	}
+
+	defer func() {
+		event.End = time.Now().Format(time.RFC1123)
+		*recordChan <- event
+	}()
+
+	setMetrics(*event)
+
+	stepLog := "Adding FADA volumes at scale to validate FADA volumes attachment in scale "
+
+	Step(stepLog, func() {
+		log.InfoD(stepLog)
+		var param = make(map[string]string)
+		var appContexts []*scheduler.Context
+
+		fadaScName := PureBlockStorageClass + time.Now().Format("01-02-15h04m05s")
+		log.Infof("Creating pure_block storage class class: %s", fadaScName)
+		param[PureBackend] = k8s.PureBlock
+		_, err := createPureStorageClass(fadaScName, param)
+		if err != nil {
+			log.Errorf("StorageClass creation failed for SC: %s", fadaScName)
+			UpdateOutcome(event, err)
+		}
+		log.InfoD("Deployng FADA based applications")
+		startTime := time.Now()
+		for x := 0; x < deploymentCount; x++ {
+			pvcName := fmt.Sprintf("%s-%d", pvcNamePrefix, x)
+			namespace := fmt.Sprintf("%s-%d", fadaNamespacePrefix, x)
+			deploymentName := fmt.Sprintf("%s-%d", fadaScName, x)
+			wg.Add(1)
+			go deployFadaApps(fadaScName, pvcName, namespace, deploymentName, &wg, &appContexts, event)
+			if x%10 == 0 {
+				// wait for previous threads to be completed
+				wg.Wait()
+			}
+		}
+		log.InfoD("Validating applications context")
+		ValidateApplications(appContexts)
```

Comment:
Can we use `validateContexts`?

Reply:
Can you add more details why not to use validateApplications here ?

Reply:
I think there is no `ginkgo.GinkgoRecover()` in this Step, so if `ValidateApplication` fails, it will panic. Also, `validateContexts` uses `UpdateOutcome` for handling errors.

Reply:
FYI:  Check this func
```

// ValidateApplications validates applications
func ValidateApplications(contexts []*scheduler.Context) {
	Step("validate applications", func() {
		log.InfoD("Validate applications")
		for _, ctx := range contexts {
			ValidateContext(ctx)
		}
	})
}
```

Reply:
I was referring to validateContexts in testTriggers.go

```go
func validateContexts(event *EventRecord, contexts *[]*scheduler.Context) {
	actualEvent := strings.Split(event.Event.Type, "<br>")[0]
	for _, ctx := range *contexts {
		stepLog := fmt.Sprintf("%s: validating app [%s]", actualEvent, ctx.App.Key)
		Step(stepLog, func() {
			log.InfoD(stepLog)
			errorChan := make(chan error, errorChannelSize)
			ctx.ReadinessTimeout = time.Minute * 10
			ValidateContext(ctx, &errorChan)
			for err := range errorChan {
				UpdateOutcome(event, err)
			}
			if strings.Contains(ctx.App.Key, fastpathAppName) {
				err := ValidateFastpathVolume(ctx, opsapi.FastpathStatus_FASTPATH_ACTIVE)
				UpdateOutcome(event, err)
			}
		})
	}
}
```

Please refer to JIRA ticket in https://github.com/portworx/torpedo/pull/2359

Reply:
Ohh I see !! Incorporated !!